src/components/navigation_mobile.tsx

```diff
@@ -0,0 +1,187 @@
+import * as React from 'react';
+import { NavLink } from 'react-router-dom';
+
+// components
+import Button from './button';
+import ProfileImage from './profile_image';
+import SlideInPanel from './slide_in_panel';
+import SubNavigationMobileMenu from './sub_navigation_mobile_menu';
+
+// components
+import { NavRoute } from './navigation';
+
+export interface NavigationMobileProps {
+    /** Details for the visible nav items */
+    visibleNavLinks: NavRoute[];
+    /** URL for the account portal */
+    manageAccountURL?: string;
+    /** Called when sign out is clicked */
+    onSignOutClickHandler?: () => void;
+    /** Called when the sign in button is clicked */
+    onSignInClickHandler?: () => void;
+    /** The label for account navigation on mobile */
+    accountButtonText?: string;
+    /** The label for sign out on mobile */
+    signOutButtonText?: string;
+    /** The label for sign in */
+    signInButtonText?: string;
+    /** The username of the user to be shown in the mobile menu */
+    accountDisplayName?: string;
+    /** The URL to the user's account page */
+    accountImageURL?: string;
+    /** Whether the user is signed out */
+    signedOut?: boolean;
+    /**
+     * Used to toggle the visibility of the mobile navigation menu.
+     * Passed in by the header component.
+     */
+    showMobileMenu?: boolean;
+    /**
+     * Called when the mobile navigation is closed.
+     * Passed in by the header component.
+     */
+    onCloseMobileMenuHandler?: () => void;
+    /** Whether the user profile should be displayed in the mobile menu */
+    showMobileAccountNav?: boolean;
+}
+
+export class NavigationMobile extends React.Component<NavigationMobileProps, {}> {
+    constructor(props) {
+        super(props);
+
+        this._onCloseMobileMenuHandler = this._onCloseMobileMenuHandler.bind(this);
+    }
+
+    private _onCloseMobileMenuHandler(): void {
+        const { onCloseMobileMenuHandler } = this.props;
+
+        if (onCloseMobileMenuHandler) {
```

Comment:
The else of this if is not covered by the tests.

Reply:
added extra test